pkgs/dart_tooling_mcp_server/lib/src/dtd.dart

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:dtd/dtd.dart';
+import 'package:json_rpc_2/json_rpc_2.dart';
+
```

Comment:
maybe file an issue and link a TODO: it might be work seing how we can share some of this code in a Dart package. DevTools implements some of the same logic to support the Flutter sidebar in VS code, which calls all these methods over DTD. Maybe at some point it would be worth creating a package dtd_editor_service or similar to share the logic. Out of scope for this PR though and something we'd only want to consider once we have more than one stable use case suggesting a reason to try to de-duplicate.

Reply:
We could put this stuff directly in the dart tooling daemon package - maybe in a different import (`vscode.dart` or similar?). I am not sure if the Intellij extension registers the exact same extensions (hopefully?).

Reply:
Both editors should implement the same extensions (although right now IntelliJ may register a subset of the ones VS Code registers, only including the ones for the property editor). Would be great to have some code generation eventually and some testing that ensures Flutter supported IDEs register the same set of functionality, especially the services we rely on for other features.